### PR TITLE
Parallel replicas: table check failover

### PR DESCRIPTION
**Pull request by devcrafter**
The PR enabled table check during parallel replicas execution and test failover in case of the table check fails/timeout

Related to [#61663](https://github.com/ClickHouse/ClickHouse/issues/61663) https://github.com/ClickHouse/ClickHouse/issues/59652
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Set desired options before CI starts or re-push after updates

#### Run only:
- [ ]  Integration tests
- [ ]  Integration tests (arm64)
- [ ]  Stateless tests (release)
- [ ]  Stateless tests (asan)
- [ ]  Stateful tests (release)
- [ ]  Stateful tests (asan)
- [ ]  No sanitizers
- [ ]  Tests with analyzer
- [ ]  Fast tests
- [ ]  Only package_debug build
- [ ]  Add your CI variant description here

#### CI options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 972af33400952abce28451f0b248ae3d0fc4579c with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61935/972af33400952abce28451f0b248ae3d0fc4579c/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/integration/test_parallel_replicas_failover/test.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        )
+        == "1\n"
+    )
+
+    assert (
+        node1.query(
+            f"SELECT ProfileEvents['ParallelReplicasUnavailableCount'] FROM system.query_log WHERE type = 'QueryFinish' AND query_id IN (SELECT query_id FROM system.query_log WHERE current_database = currentDatabase() AND log_comment = '{log_comment}' AND type = 'QueryFinish' AND initial_query_id = query_id)  SETTINGS allow_experimental_parallel_reading_from_replicas=0"
+        )
+        == "1\n"
+    )
```
Comment (nickitat) [inline tests/integration/test_parallel_replicas_failover/test.py:97]: I'd merge these into a single query

**Comment by nickitat** [comment]
sorry for a silly question. for me this patch does an equivalent transformation, which part of the diff exactly fixes "Query with parallel replicas fails if replica is unresponsive during handshake"?

**Comment by devcrafter** [comment]
> sorry for a silly question. for me this patch does an equivalent transformation, which part of the diff exactly fixes "Query with parallel replicas fails if replica is unresponsive during handshake"?

It doesn't fix it. At first, I thought that a timeout for table check could be a reason. It turns out that it's not. But now we have a test with the timeout for table check.

**Review by nickitat — APPROVED** [review]

**Comment by devcrafter** [comment]
AST Fuzzer - https://github.com/ClickHouse/ClickHouse/issues/63023

**Pull request merged**